mlflow/sklearn/__init__.py

```diff
@@ -948,6 +1334,21 @@ def patched_fit(original, self, *args, **kwargs):
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
 
+        for func_name in ["predict", "transform", "predict_proba"]:
+            if hasattr(class_def, func_name) and callable(getattr(class_def, func_name)):
+                safe_patch(
+                    FLAVOR_NAME, class_def, func_name, patched_predict, manage_run=False,
+                )
+
+        if hasattr(class_def, 'score') and callable(class_def.score):
+            safe_patch(
+                FLAVOR_NAME, class_def, 'score', patched_model_score, manage_run=False,
+            )
+
+    from sklearn import metrics
+    for metric_name in _get_metric_name_list():
+        safe_patch(FLAVOR_NAME, metrics, metric_name, patched_metric_api, manage_run=False)
```

Comment:
I think there may be a small bug in the metric name that we're using in the patched function. I tried computing 3 different metrics in the following example:

```
import sklearn

from sklearn import datasets
from sklearn.linear_model import LogisticRegression

import mlflow
mlflow.autolog()

iris = datasets.load_iris()

logistic = LogisticRegression(solver="saga", max_iter=200, random_state=0)
train_out = logistic.fit(iris.data, iris.target)

from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score

test_y = logistic.predict(iris.data)

r2_score(test_y, iris.target)
mean_absolute_error(test_y, iris.target)
mean_squared_error(test_y, iris.target)
```

However, this created 3 metrics with incorrect names:

```
brier_score_loss_unknown_dataset
brier_score_loss-2_unknown_dataset
brier_score_loss-3_unknown_dataset
```

Perhaps `brier_score_loss` is the first metric name that appears in the list?

Reply:
addressed. it is code bug.